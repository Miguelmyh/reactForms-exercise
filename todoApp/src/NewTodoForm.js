import React, { useState } from "react";

const NewTodoForm = ({ addTodo }) => {
  const INITIAL_STATE = {
    task: "",
  };
  const [formData, setFormData] = useState(INITIAL_STATE);
  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({ ...formData });
    setFormData(INITIAL_STATE);
  };
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((data) => ({ ...data, [name]: value }));
  };

  return (
    <form onSubmit={handleChange}>
      <label htmlFor="todo">Todo: </label>
      <input
        id="todo"
        name="task"
        placeholder="TodoItem"
        value={formData.task}
        onChange={handleChange}
      />
      <button onClick={handleSubmit}>Submit</button>
    </form>
  );
};

export default NewTodoForm;
