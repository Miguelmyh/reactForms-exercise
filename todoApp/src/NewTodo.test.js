import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

test("should render", () => {
  render(<NewTodoForm />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

test("should have form fields", () => {
  const { queryByPlaceholderText } = render(<NewTodoForm />);
  const input = queryByPlaceholderText("TodoItem");
  expect(input).toBeInTheDocument();
});
