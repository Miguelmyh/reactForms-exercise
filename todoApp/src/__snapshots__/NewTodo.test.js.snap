// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Todo: 
    </label>
    <input
      id="todo"
      name="task"
      placeholder="TodoItem"
      value=""
    />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
